### PR TITLE
ProcessRuntime autodeploy process behaviour too aggressive

### PR DESCRIPTION
**Pull request by daisuke-yoshimoto**
## Issue

* Refer to the following issue
  * https://github.com/Activiti/Activiti/issues/2138

### fail on the first error

* Condition
  * `spring.activiti.deploymentMode=default` or no setting

* Behavior
  * The Runtime doesn't start if there is at least one error while deploying process definitions.
  * This is default behavior.

### fail on no process definition

* Condition
  * `spring.activiti.deploymentMode=fail-on-no-process`

* Behavior
  * The Runtime start if there is at least a process definition deployed.

### never fail (for the case where the application might be providing other functionality besides process endpoints)

* Condition
  * `spring.activiti.deploymentMode=never-fail`

* Behavior
  * The Runtime start even if there is no process definitions deployed.

### CODE REVIEW

File: activiti-spring/src/test/java/org/activiti/spring/test/autodeployment/NeverFailAutoDeploymentStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    private void cleanUp() {
+        List<org.activiti.engine.repository.Deployment> deployments = repositoryService.createDeploymentQuery().list();
+        for (org.activiti.engine.repository.Deployment deployment : deployments) {
+            repositoryService.deleteDeployment(deployment.getId(), true);
+        }
+    }
+
+    @Test
+    public void testValidResources() {
+        cleanUp();
```
[the remark this answers is not in the document]
Comment (daisuke-yoshimoto) [reply activiti-spring/src/test/java/org/activiti/spring/test/autodeployment/NeverFailAutoDeploymentStrategyTest.java]: @igdianov 
Sorry. It seems that annotation can not be used due to the influence of junit.framework.TestCase class inheritance relationship.

So I fixed to override the setUp/tearDown method.

**Comment by daisuke-yoshimoto** [comment]
@igdianov 
Thanks. Could you change the review status?

**Comment by daisuke-yoshimoto** [comment]
@salaboy Please review this pr when you have time.

File: .mergify.yml [hunk 2]
```diff
@@ -12,13 +12,17 @@ pull_request_rules:
       merge:
         method: merge
         strict: true
-  - name: automatic merge when CI passes and 1 approving reviews
+  - name: automatic merge when CI passes and all requested reviews approved
```
Comment (salaboy) [inline .mergify.yml]: @daisuke-yoshimoto I would remove the changes in this file.. we are trying to align all repos to look the same. Is there any reason why you changed the defaults?

**Pull request merged**